### PR TITLE
Update on Fence OP

### PR DESCRIPTION
**Pull request by Jianhui-Li**
Please review these guidelines to help with the review process:
- [ ] Have you provided a meaningful PR description?
- [ ] Have you added a test, a reproducer, or a reference to an issue with a reproducer?
- [ ] Have you tested your changes locally for CPU and GPU devices?
- [ ] Have you made sure that new changes do not introduce compiler warnings?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?
- [ ] Have you organized your commits logically and ensured each can be built by itself?

### CODE REVIEW

File: docs/rfcs/XeGPU.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ Below is a summary.
 |init_nbarrier	| operation ::= XeGPU.init_nbarrier $nbarrier_id, $participant_thread_num attr-dict : Uint8_t, Uint8_t -> type($nbarrier) | %nbarrier = XeGPU.alloc_nbarrier %nbarrier_id, %participant_thread_num : Uint8_t, Uint8_t -> !XeGPU.nbarrier |
 |nbarrier_arrive	| operation ::= XeGPU.nbarrier_arrive $nbarrier : type($nbarrier) | XeGPU.nbarrier_arrive %nbarrier : !XeGPU.nbarrier |
 |nbarrier_wait	| operation ::= XeGPU.nbarrier_wait $nbarrier : type($nbarrier) | XeGPU.nbarrier_wait %nbarrier : !XeGPU.nbarrier |
-|Mfence	| operation ::= XeGPU.mfence attr-dict | XeGPU.mfence {fence_scope = global} |
+|fence	| operation ::= XeGPU.fence attr-dict | XeGPU.mfence {scope = gpu, memory_kind = global} |
```
Comment (Garra1980) [inline docs/rfcs/XeGPU.md:29]: let's remove mfence completely?

Comment (Jianhui-Li) [reply docs/rfcs/XeGPU.md]: I changed mfence to fence.

File: docs/rfcs/XeGPU.md [hunk 2]
```diff
@@ -271,19 +271,18 @@ alloc_nbarrier allocates a set of named barriers with the specified number. Name
...
-Attribute `Memory_kind` describes the memory kind. "ugm" means the global memory, "slm" means the share local memory.
+Attribute `scope` describes the scope of fence. "workgroup" means that the scope is within each work group. "gpu" means the scope is across work groups within the gpu.
+Attribute `Memory_kind` describes the memory kind. "global" means the global memory, "shared" means the shared local memory.
 
 `compile_hint` passes performance hints to the lower-level compiler. The schedule_barrier hint prevents instructions from being reordered by a lower-level compiler. For example, a prefetch instruction is location-sensitive, but the lower-level compiler may schedule it to an undesired location.  
 ```mlir  
 XeGPU.compile_hint {hint=schedule_barrier}
 ```
-nbarrrier, mfence, and compile_hint operations lower to uniform instructions, so there is no need to specify the sg_map or VC mode.
+nbarrrier, fence, and compile_hint operations lower to uniform instructions, so there is no need to specify the sg_map or VC mode.
```
Comment (Garra1980) [inline docs/rfcs/XeGPU.md:285]: Typo: rrr->rr in nbarrier

**Review by Garra1980 — APPROVED** [review]

**Review by silee2 — APPROVED** [review]

File: docs/rfcs/XeGPU.md (hunk 2 above)
Comment (silee2) [reply docs/rfcs/XeGPU.md]: Fixed.

**Pull request merged**